lib/pharos/cluster_manager.rb

```diff
@@ -133,23 +133,16 @@ def apply_phases
       apply_phase(Phases::ConfigureTelemetry, [master_hosts.first], master: master_hosts.first)
     end
 
-    def apply_reset_hosts(hosts, drain: true, delete: true)
+    # @param hosts [Array<Pharos::Configuration::Host>]
+    def apply_reset_hosts(hosts)
       master_hosts = sorted_master_hosts
-      apply_phase(Phases::GatherFacts, hosts, ssh: true, parallel: true)
-
-      if drain || delete
-        apply_phase(Phases::ConfigureClient, [master_hosts.first], ssh: true, master: master_hosts.first, parallel: false, optional: true)
+      if master_hosts.first.master_sort_score == 0
```

Comment:
`master_hosts.first.checks['api_healthy']` (or a helper method) could be clearer.